### PR TITLE
feat: add `dt.to_string` to `DaskExpr`

### PR DESCRIPTION
**Pull request by lucianosrp**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #637  
- Closes #

## Checklist

- [x] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

## If you have comments or can explain your changes, please do so below.

~Dask seems to be failing for `%.f`, any suggesstions on how to handle it ?~ Solved

### CODE REVIEW

File: tests/expr_and_series/dt/to_string_test.py [hunk 1]
```diff
@@ -17,31 +18,73 @@
...
 @pytest.mark.skipif(is_windows(), reason="pyarrow breaking on windows")
-def test_dt_to_string(constructor_eager: Any, fmt: str) -> None:
+def test_dt_to_string_series(constructor_eager: Any, fmt: str) -> None:
     input_frame = nw.from_native(constructor_eager(data), eager_only=True)
     input_series = input_frame["a"]
 
     expected_col = [datetime.strftime(d, fmt) for d in data["a"]]
 
-    result = input_series.dt.to_string(fmt).to_list()
+    result = input_frame.select(input_series.dt.to_string(fmt))
```
Comment (FBruzzesi) [inline tests/expr_and_series/dt/to_string_test.py:37]: Somewhat nitpick:

```suggestion
    result = {"a": input_series.dt.to_string(fmt)}
```

File: tests/expr_and_series/dt/to_string_test.py [hunk 2]
```diff
@@ -17,31 +18,73 @@
...
         # the fraction of a second.
-        result = [x[: x.find(".")] if "." in x else x for x in result]
-    assert result == expected_col
-    result = input_frame.select(nw.col("a").dt.to_string(fmt))["a"].to_list()
-    if any(
-        x in str(constructor_eager) for x in ["pandas_pyarrow", "pyarrow_table", "modin"]
-    ):
+        result = input_frame.select(
+            input_series.dt.to_string(fmt).str.replace(r"\.\d+$", "")
+        )
```
Comment (FBruzzesi) [inline tests/expr_and_series/dt/to_string_test.py:46]: Same as above:

```suggestion
        result = {"a": input_series.dt.to_string(fmt).str.replace(r"\.\d+$", "")}
```

**Review by FBruzzesi — COMMENTED** [review]
That's definitly more effort than expected! Thanks for putting the work on this 🙌🏼
I left only a nitpick comment on how to test/compare series values

**Comment by lucianosrp** [comment]
@FBruzzesi thanks as always, could you check now?

**Comment by lucianosrp** [comment]
I am not able to replicate [this failure](https://github.com/narwhals-dev/narwhals/actions/runs/10440899295/job/28911371149)  in my local, any suggestions ?

**Comment by FBruzzesi** [comment]
> I am not able to replicate [this failure](https://github.com/narwhals-dev/narwhals/actions/runs/10440899295/job/28911371149) in my local, any suggestions ?

Sharing what I am discovering along the way: replacing `with_columns` with `select` does not raise an error. I am a bit lost as well 🙈

What I mean exactly is:

```py
result = nw.from_native(df).select(
    nw.col("a").dt.to_string("%Y-%m-%d")
)
```

**Comment by lucianosrp** [comment]
> replacing with_columns with select does not raise an error

Do you think there might be an issue with Narwhals' internals ? Or just Modin? 
Shall we just change the test and monitor if we see this moving forward? 😓

**Comment by FBruzzesi** [comment]
I would not assume is something bad we are doing because for pandas there are no issues and the logic should be the same. Apart from that, I never worked with modin directly 🥲

@MarcoGorelli would you mind weighting in for this whenever you have the time?

**Comment by MarcoGorelli** [comment]
thanks for the ping! will take a look at this and other open prs later today 😇

**Comment by MarcoGorelli** [comment]
looks like a bug in Modin - I've reported it to them here https://github.com/modin-project/modin/issues/7371

shall we just xfail this test for modin for now?

**Review by MarcoGorelli — APPROVED** [review]
thanks @lucianosrp and @FBruzzesi for reviewing

I think you also need to xfail `test_dt_to_string_iso_local_datetime_expr` - then it should be good to go, feel free to merge (just check that all tests, apart from pyarrow-nightly, pass)

**Comment by lucianosrp** [comment]
Thanks @FBruzzesi  and @MarcoGorelli as always 🫡

**Pull request merged**